src/symbols-count.js

```diff
@@ -1,71 +1,95 @@
-import sketch from 'sketch'
+import Sketch from 'sketch';
+import path from '@skpm/path';
+import FormData from 'sketch-polyfill-fetch/lib/form-data';
 
-const doc = sketch.getSelectedDocument()
-const pages = doc.pages
+import {SECRET_NAME, TOKEN_NAME, API_ENDPOINT, CLIENT_ID} from './config';
+import {refresh} from './authenticate';
+import track from './track';
 
-const ignoreTypes = [
-  'Page',
-  'Artboard',
-  'Group',
-  'SymbolMaster'
-]
+const doc = Sketch.getSelectedDocument();
+const pages = doc.pages;
 
-export default function() {
-  let layerCount = 0
-  let symbolsCount = 0
+const ignoreTypes = ['Page', 'Artboard', 'Group', 'SymbolMaster'];
+
+export default async function onRun() {
```

Comment:
does this actually run it async? without blocking the ui at all? i tried to get this to work ages ago but i don't think it's supported properly because of the way sketch runs plugins on the single main thread...

Reply:
It seems to be working within the context of the plugin. The main sketch UI I think will still be blocked, but it's a useful pattern for some of the authentication functions where we want to wait for a token to be returned.